BayAreaMetro/bayarea_urbansim
# Preserve units as deed-restricted

The changes add a step to the model that preserves affordable housing by tagging units as deed-restricted. It has no relationship to price which is all calculated off-model. It works with code that only allows Q1 households in deed-restricted units. 

The strategy starts being applied in 2015, the first "simulation" year (and the first year development projects are added). The model flow now goes as follows:
- "Affordable" units exist in the base data, these can be redeveloped if/when the building is > 20 years old
-  Development projects are built, including "affordable" units, these locations can't be redeveloped
-  "Affordable" units are preserved within buildings, these locations can be redeveloped if/when the buildings is > 20 years old
- "Affordable" units are produced in market rate buildings, these can be redeveloped once they're > 20 years old

I added the new deed-restricted units to the deed-restricted units column of the residential units and buildings tables. I also added a `preserved_units` column to the buildings table to summarize this strategy independently. I added "gg_id" to the parcels_geography import since there is a Draft Blueprint and Final Blueprint version of the variable now. 

To test the strategy:
I checked that the simulation completes with no errors. To make sure the code details work correctly, I checked that when the filters are applied, the correct units are marked as deed-restricted (have not re-checked with the new geographies but the totals indicate that they're working). I also checked that the marked units continue to add to the set of deed-restricted over the simulation in the residential units and buildings tables. 

In the bigger picture, I stepped through the output to make sure that Q1 households are being placed in the deed-restricted units (confirm this feature). 
Currently, the breakdown of income levels in deed-restricted units is:
1.0    556,600
4.0      62,488
2.0      30,782
3.0      29,227
The breakdown of income levels in non-deed-restricted units is:
4.0    1,266,673
2.0       75,7223
3.0       75,4425
1.0       58,5894
The non-Q1 in deed-restricted units should be in part due to non-Q1 households in units marked deed-restricted in the base year, who never moved out for a Q1 households to move in. But I'll continue to look into this. 

I also checked whether marking these units as deed-restricted didn't created any unplaced households, which they did not.  Then, I checked whether the `no_unplaced` models were being used, in other words if the new set of deed-restricted units was creating unplaced households due to tenure. Currently: 9K and 6K households are placed in 2045 and 2050 using this model.
